.github/workflows/maven.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Maven, and cache/restore any dependencies to improve the workflow execution time
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  push:
```

Comment:
Since we restrict pushes to main so that they have to go through a PR is the push to main here necessary? It will result in a double build: one that happened on the PR and one that happens on the merge.

Reply:
Removed the `on push`